edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookExternalAssessmentServiceImpl.java

```diff
@@ -271,23 +271,20 @@ public void removeExternalAssessment(final String gradebookUid,
         // any deletions at present. See the comments to deleteGradebook
         // for the details.
         HibernateTemplate hibTempl = getHibernateTemplate();
-        
-        List toBeDeletedEvents = hibTempl.find("from GradingEvent as ge where ge.gradableObject=?", asn);
-        int numberDeletedEvents = toBeDeletedEvents.size();
-        hibTempl.deleteAll(toBeDeletedEvents);
-        if (logData.isDebugEnabled()) logData.debug("Deleted " + numberDeletedEvents + "records from gb_grading_event_t");
-
-        List toBeDeleted = hibTempl.find("from AssignmentGradeRecord as agr where agr.gradableObject=?", asn);
-        int numberDeleted = toBeDeleted.size();
-        hibTempl.deleteAll(toBeDeleted);
-        if (log.isInfoEnabled()) log.info("Deleted " + numberDeleted + " externally defined scores");
-
-        toBeDeleted = hibTempl.find( "from Comment as c where c.gradableObject = ?", asn );
-        hibTempl.deleteAll( toBeDeleted );
-        if( log.isInfoEnabled() )
-        {
-            log.info( "Deleted " + toBeDeleted.size() + " externally defined score comments" );
-        }
+
+        hibTempl.execute(new HibernateCallback() {
+            public Object doInHibernate(Session session) {
+                int numDeleted = session.createQuery("delete GradingEvent where gradableObject=:go").setParameter("go", asn).executeUpdate();
+                logData.debug("Deleted " + numDeleted + " records from gb_grading_event_t");
+
+                numDeleted = session.createQuery("delete AssignmentGradeRecord where gradableObject=:go").setParameter("go", asn).executeUpdate();
+                log.info("Deleted " + numDeleted + " externally defined scores");
```

Comment:
Top of the file says:
```
private static final Log log = LogFactory.getLog(GradebookExternalAssessmentServiceImpl.class);
// Special logger for data contention analysis.
private static final Log logData = LogFactory.getLog(GradebookExternalAssessmentServiceImpl.class.getName() + ".GB_DATA");
```
So different log names for some reason. Unless better documented I would just remove the logData one and consolidate.